csvcubed/csvcubed/readers/configdeserialiser.py

```diff
@@ -0,0 +1,572 @@
+                "type": column_type,
+                "value": column_name,
+                "datatype": "decimal",
+            }
+
+        elif column_type in ["measures", "units"]:
+            column_dict = {"type": column_type, "new": True}
+        elif column_type == "attribute":
+            # Note: attribute type columns are currently not supported
+            raise ValueError(
+                "Column type 'Attribute' is not supported when using csv naming convention"
+            )
+
+        else:
+            raise ValueError(f"Column type '{column_type}' is not supported.")
+
+        # As all columns will be new in this scenario get a pd.Series for the column
+
+        try:
+            qb_column = map_column_to_qb_component(
+                column_title=column_name,
+                column=column_dict,
+                data=column_data.astype("category"),
+                info_json_parent_dir=None,
+            )
+            cube_columns.append(qb_column)
+
+        except Exception as err:
+            log.error(f"{type(err)} exception raised because: {repr(err)}")
+
+    cube_metadata = _metadata_from_dict(
+        {"id": str(csv_path).upper(), "title": csv_path.name.upper().replace("_", " ")}
+    )
+    # TODO - Create the new Cube
+    cube = Cube(metadata=cube_metadata, data=data, columns=cube_columns)
+    # validation_errors = cube.validate()
+    validation_errors = cube.pydantic_validation()
+
+    return cube, validation_errors
+
+
+def get_cube_from_info_json(
+        config_path: Path,
+        data: pd.DataFrame,
+        cube_id: Optional[str] = None,
+        log: logging.RootLogger = None
+) -> Tuple[QbCube, List[ValidationError]]:
+    """
+    Generates a Cube structure from a config.json input.
+    :return: tuple of cube and json schema errors (if any)
+    """
+
+    config = _load_resource(config_path)
+    schema_path = Path("..", "..", "..", "..", "csvcubed", "schema", "cube-config-schema.json").resolve()
+    schema = _load_resource(schema_path)
+    try:
+        errors = validate_dict_against_schema(value=config, schema=schema, log=log)
+        if errors:
+            return None, errors
+
+        if cube_id is not None:
+            config = _override_config_for_cube_id(config, cube_id)
+
+        if config is None:
+            raise Exception(f"Config not found for cube with id '{cube_id}'")
+
+    except Exception as err:
+        print(err)
+
+    return _from_info_json_dict(config, data, config_path.parent), None
+
+
+def build(
+    csv_path: Path,
+    info_json: Optional[Path] = None,
+    # catalog_metadata_json_file: Optional[Path]
+    # output_directory: Path,
+    # csv_path: Path,
+    # fail_when_validation_error_occurs: bool,
+    # validation_errors_file_out: Optional[Path],
+):
+    # print(f"{Style.DIM}CSV: {csv_path.absolute()}")
+    # print(f"{Style.DIM}info.json: {info_json.absolute()}")
+    data: pd.DataFrame = pd.read_csv(csv_path)
+
+
+    if info_json:
```

Comment:
We need this to allow conventional column definitions at the same time as a user providing a `config.json`.